bin/gen_motd.sh

```diff
@@ -12,14 +12,15 @@ source "$(cd $(dirname "${BASH_SOURCE[0]}") &>/dev/null && pwd)/common.sh"
     echo "   MOTD generated $(date)"
     echo '   (Psst, write up to 24 bytes in a file called ~/motd)'
     echo
-    find /home -maxdepth 2 -type f -name motd | sort | while read motd_path; do
+    find /home -mindepth 2 -maxdepth 2 -type f -name motd | sort | while read motd_path; do
         user=$(echo $motd_path | cut -d/ -f3)
         motd=$(
-            { timeout 1 cat $motd_path || true; } | \
-            tr '[:space:]' " " | \
-            sed -e 's/[[:space:]]\+/ /g' | \
-            tr -cd '[:graph:][:space:]' | \
-            cut -c1-24 \
+            timeout 1 sed -E \
+                -e 's/[^[:graph:][:space:]]//g' \
+                -e 's/[[:space:]]+/ /g' \
+                -e 's/^(.{0,24}).*$/\1/' \
+                $motd_path \
+            || true
         )
         printf "%${max_uname_len}s: %-24s\n" "$user" "$motd"
```

Comment:
```suggestion
        printf "%${max_uname_len}s: %-24.*s\n" "$user" 24 "$motd"
```

With this change I agree we can then remove the `-e 's/^(.{0,24}).*$/\1/' \` line.

Reply:
Indeed, I overlooked that `%-24s` alignes, but not cuts.

Then, I am new to asterisk here - never encountered before - and while researching it, discovered this:

https://www.man7.org/linux/man-pages/man1/printf.1p.html

>        Note that this printf utility, like the printf() function defined
>       in the System Interfaces volume of POSIX.1‐2017 on which it is
>       based, makes no special provision for dealing with multi-byte
>       characters when using the %c conversion specification or when a
>       precision is specified in a %b or %s conversion specification.
>       Applications should be extremely cautious using either of these
>       features when there are multi-byte characters in the character
>       set.

Although bash uses different `printf` - a built-in one, it doesn't count multibyte characters properly too, same as `/usr/bin/printf`
```
> printf "%2.*s\n" 2 "٩(^‿^)۶"  # expecting "٩("
٩
```

It turns out, we can't remove the `-e 's/^(.{0,24}).*$/\1/' \` line.
We need to add another line that would format motd, like `-e ':a; s/^.{0,23}$/& /; ta'`,
and remove formatting in printf for motd, as it would already be formatted.

---

`:a` - define label `a`
`s/^.{0,23}$/& /` - insert one space after motd if length of motd is <= 23
`ta` - repeat from label `a`